### PR TITLE
SSL Certificate checking

### PR DESCRIPTION
**Pull request by av2k**
Hello,

this is the code we use since about 9 months for certificate checking.

It adds a new column in the database to store the minimum days of remaining certificate validity and uses OpenSSL to get the server certificate details.

### CODE REVIEW

**Comment by jimwebb** [comment]
This patch worked well for me, with two small issues:

- I believe it performs the SSL certificate check for all domains, even HTTP ones (which will fail)
- Phpservermon has a related issue in which ports can't be changed after a server is entered (https://github.com/phpservermon/phpservermon/issues/418). Normally this isn't a problem and the server monitor works fine if the port is inaccurate, but this patch needs those ports to be accurate.

(Also, if someone is here trying to implement this for themselves, you'll need to modify the "servers" database table with two new fields for this patch to work, as described in the code of this PR; they don't automatically get created and the list of servers will "disappear" without an error message if those tables don't exist.)

Thanks for doing this!

**Comment by TimZ99** [comment]
The monitor marks the websites as offline when the SSL certificate is still valid, but just under the number of days, you specify. I would say that a warning is enough. Making the box yellow instead of red and still monitoring the latency could be a good solution.

File: src/includes/functions.inc.php [hunk 1]
```diff
@@ -355,6 +355,21 @@ function psm_curl_get($href, $header = false, $body = true, $timeout = null, $ad
 	return $result;
 }
 
+/**
+ * Get SSL certificate information about the peer 
+ *
+ * @param string $href
+ * @return array
+ */
+function psm_cert_info_get($href, $port) {
```
Comment (TimZ99) [inline src/includes/functions.inc.php:364]: Feedback in the conversation.

**Review by TimZ99 — CHANGES_REQUESTED** [review]

**Comment by DominikWA** [comment]
Is it foreseeable when this will become part of the phpservermonitor? A web page without a valid certificate will appear to users as if it were unreachable, so the server monitor should report it.

**Comment by TimZ99** [comment]
@DominikWA  I think I will put it in v3.5.

**Pull request closed without merging**